src/printer.rs

```diff
@@ -96,6 +96,41 @@ impl<'a> Printer<'a> {
                     )?
                 );
             }
+            SerializationFormat::Junit => {
+                use quick_junit::{NonSuccessKind, Report, TestCase, TestCaseStatus, TestSuite};
+
+                // Group by filename.
+                let mut grouped_messages = BTreeMap::default();
+                for message in &diagnostics.messages {
+                    grouped_messages
+                        .entry(&message.filename)
+                        .or_insert_with(Vec::new)
+                        .push(message);
+                }
+
+                let mut report = Report::new("ruff");
+                for (filename, messages) in grouped_messages {
+                    let mut test_suite = TestSuite::new(filename);
+                    test_suite
+                        .extra
+                        .insert("package".to_string(), "org.ruff".to_string());
+                    for message in messages {
+                        let mut status = TestCaseStatus::non_success(NonSuccessKind::Failure);
+                        status.set_message(message.kind.body());
+                        let mut case =
+                            TestCase::new(format!("org.ruff.{}", message.kind.code()), status);
+                        case.set_classname(filename);
```

Comment:
The ESLint version strips the `.js` suffix, so maybe we should do that? I'm not sure.

Reply:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="ruff" tests="4" failures="4" errors="0">
    <testsuite name="/Users/messense/Projects/ruff/resources/test/fixtures/A001.py" tests="4" disabled="0" errors="0" failures="4" package="org.ruff">
        <testcase name="org.ruff.F401" classname="/Users/messense/Projects/ruff/resources/test/fixtures/A001" line="1" column="8">
            <failure message="`some` imported but unused">/Users/messense/Projects/ruff/resources/test/fixtures/A001.py:1:8: `some` imported but unused</failure>
        </testcase>
        <testcase name="org.ruff.F401" classname="/Users/messense/Projects/ruff/resources/test/fixtures/A001" line="2" column="18">
            <failure message="`some.other` imported but unused">/Users/messense/Projects/ruff/resources/test/fixtures/A001.py:2:18: `some.other` imported but unused</failure>
        </testcase>
        <testcase name="org.ruff.F821" classname="/Users/messense/Projects/ruff/resources/test/fixtures/A001" line="5" column="12">
            <failure message="Undefined name `annotation`">/Users/messense/Projects/ruff/resources/test/fixtures/A001.py:5:12: Undefined name `annotation`</failure>
        </testcase>
        <testcase name="org.ruff.F841" classname="/Users/messense/Projects/ruff/resources/test/fixtures/A001" line="18" column="1">
            <failure message="Local variable `ValueError` is assigned to but never used">/Users/messense/Projects/ruff/resources/test/fixtures/A001.py:18:1: Local variable `ValueError` is assigned to but never used</failure>
        </testcase>
    </testsuite>
</testsuites>
```